src/main/java/org/mockito/verification/VerificationWrapperInOrderWrapper.java

```diff
@@ -0,0 +1,49 @@
+package org.mockito.verification;
+
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.internal.InOrderImpl;
+import org.mockito.internal.verification.InOrderWrapper;
+import org.mockito.internal.verification.VerificationOverTimeImpl;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+
+public class VerificationWrapperInOrderWrapper implements VerificationMode {
+    private final VerificationMode delegate;
+
+    public VerificationWrapperInOrderWrapper(VerificationWrapper<?> verificationWrapper, InOrderImpl inOrder) {
+        VerificationMode verificationMode = verificationWrapper.wrappedVerification;
+
+        VerificationMode inOrderWrappedVerificationMode = wrapInOrder(verificationWrapper, verificationMode, inOrder);
+
+        delegate = verificationWrapper.copySelfWithNewVerificationMode(inOrderWrappedVerificationMode);
+    }
+
+    @Override
+    public void verify(VerificationData data) {
+        delegate.verify(data);
+    }
+
+    @Override
+    public VerificationMode description(String description) {
+        return null;
+    }
+
+    private VerificationMode wrapInOrder(VerificationWrapper<?> verificationWrapper, VerificationMode verificationMode, InOrderImpl inOrder) {
+        if (verificationMode instanceof VerificationInOrderMode) {
+            final VerificationInOrderMode verificationInOrderMode = (VerificationInOrderMode)verificationMode;
+            return new InOrderWrapper(verificationInOrderMode, inOrder);
+        } else if (verificationMode instanceof VerificationOverTimeImpl) {
+            final VerificationOverTimeImpl verificationOverTime = (VerificationOverTimeImpl)verificationMode;
+            if (verificationOverTime.isReturnOnSuccess()) {
+                return new VerificationOverTimeImpl(verificationOverTime.getPollingPeriodMillis(),
```

Comment:
What I wanted to say is that the parameters passed to the constructor should be assigned to local variables first. This avoids interleaved calls for the construction and improves clarity.

```
private static VerificationOverTime wrapVerificationOverTime(VerificationOverTime verificationOverTime, VerificationWrapper verificationWrapper, InOrder inOrder){
    long pollPeriodeMillis =verificationOverTime.getPollingPeriodMillis();
    long durationMillis = verificationOverTime.getTimer().duration();
    InOrder warppedInOrder=wrapInOrder(verificationWrapper, verificationOverTime.getDelegate(), inOrder);
    boolean returnOnSuccess= verificationOverTime.isReturnOnSuccess();

    return new VerificationOverTimeImpl(pollPeriodeMillis, durationMillis, wrappedInOrder, returnOnSuccess);
}
```